packages/web3/test/e2e/sepolia/get_past_logs.test.ts

```diff
@@ -0,0 +1,127 @@
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { FMT_BYTES, FMT_NUMBER, bytesToBuffer, hexToBytes } from 'web3-utils';
+
+import Web3, { LogAPI } from '../../../src';
+import {
+	closeOpenConnection,
+	getSystemTestBackend,
+} from '../../shared_fixtures/system_tests_utils';
+import { toAllVariants } from '../../shared_fixtures/utils';
+import { getSystemE2ETestProvider } from '../e2e_utils';
+
+describe(`${getSystemTestBackend()} tests - getPastLogs`, () => {
+	const provider = getSystemE2ETestProvider();
+	const expectedLogs: LogAPI[] = [
+		{
+			address: '0xedfd52255571b4a9a9d4445989e39f5c14ff0447',
+			blockHash: '0xdb1cb1fc3867fa28e4ba2297fbb1e65b81a3212beb1b73cbcbfe40c4192ee948',
+			blockNumber: '0x314675',
+			data: '0x',
+			logIndex: '0x4',
+			removed: false,
+			topics: [
+				'0x342827c97908e5e2f71151c08502a66d44b6f758e3ac2f1de95f02eb95f0a735',
+				'0x0000000000000000000000000000000000000000000000000000000000000000',
+				'0x000000000000000000000000a127c5e6a7e3600ac34a9a9928e52521677e7211',
+			],
+			transactionHash: '0xe3d28c23ffcd8b94b8f2fd802b4c8dd17a5f19992660acd082ac229ab410b959',
+			transactionIndex: '0x8',
+		},
+	];
+
+	let web3: Web3;
+
+	beforeAll(() => {
+		web3 = new Web3(provider);
+	});
+
+	afterAll(async () => {
+		await closeOpenConnection(web3);
+	});
+
+	it.each(
+		toAllVariants<{
+			format: string;
+		}>({
+			format: Object.values(FMT_BYTES),
+		}),
+	)('should getPastLogs for deployed contract', async ({ format }) => {
+		const result = await web3.eth.getPastLogs(
+			{
+				fromBlock: 'earliest',
+				toBlock: 'latest',
+				address: '0xEdFd52255571b4a9A9d4445989E39f5c14Ff0447',
+			},
+			{
+				number: FMT_NUMBER.HEX,
+				bytes: format as FMT_BYTES,
+			},
+		);
+
+		switch (format) {
+			case 'BYTES_HEX':
+				// eslint-disable-next-line jest/no-conditional-expect
+				expect(result).toStrictEqual(expectedLogs);
+				break;
+			case 'BYTES_BUFFER':
+				// eslint-disable-next-line jest/no-conditional-expect
+				expect(result).toStrictEqual([
+					{
+						...expectedLogs[0],
+						blockHash: bytesToBuffer(hexToBytes(expectedLogs[0].blockHash as string)),
+						data: bytesToBuffer(hexToBytes(expectedLogs[0].data as string)),
+						transactionHash: bytesToBuffer(
+							hexToBytes(expectedLogs[0].transactionHash as string),
+						),
+						topics: expectedLogs[0].topics?.map(topic =>
+							bytesToBuffer(hexToBytes(topic)),
+						),
+						// TODO Should these be formatted?
+						// blockNumber: bytesToBuffer(hexToBytes((expectedLogs[0]).blockNumber as string)),
+						// data: bytesToBuffer(hexToBytes((expectedLogs[0]).data as string)),
+						// logIndex: bytesToBuffer(hexToBytes((expectedLogs[0]).logIndex as string)),
+						// transactionIndex: bytesToBuffer(hexToBytes((expectedLogs[0]).transactionIndex as string)),
```

Comment:
I feel like these properties were missed for formatting, should we create an issue to fix this @jdevcs?

Reply:
sure.

Reply:
#6027 tracks this